x/swap/keeper/deposit.go

```diff
@@ -0,0 +1,53 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/kava-labs/kava/x/swap/types"
+)
+
+func (k Keeper) Deposit(ctx sdk.Context, depositor sdk.AccAddress, amountA sdk.Coin, amountB sdk.Coin) error {
+	poolName := types.PoolName(amountA.Denom, amountB.Denom)
+
+	_, found := k.GetPool(ctx, poolName)
+	if found {
+		return sdkerrors.Wrap(types.ErrNotImplemented, fmt.Sprintf("can not deposit into existing pool '%s'", poolName))
+	}
+
+	// TODO: extract method
+	params := k.GetParams(ctx)
+	creationAllowed := false
+	for _, p := range params.AllowedPools {
+		if p.TokenA == amountA.Denom && p.TokenB == amountB.Denom {
+			creationAllowed = true
+		}
+	}
+	if !creationAllowed {
+		return sdkerrors.Wrap(types.ErrNotAllowed, fmt.Sprintf("can not create pool '%s'", poolName))
+	}
+
+	// TODO: extract method, wrap error
```

Comment:
Not convinced that we need to extract this into its own method

Reply:
I've left this as a TODO in https://github.com/Kava-Labs/kava/pull/930

Reply:
👍 sounds good